wc_mugdha/main.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+)
+
+var (
+	lineCount int
+	charCount int
+	wordCount int
+	wordEnd   uint8
+)
+
+func main() {
+	allArgs := os.Args[1:]
+
+	for _, fileName := range allArgs {
+		lineCount, wordCount, charCount = 0, 0, 0
+		f, err := os.Open(fileName)
+		if err != nil {
+			erroredExit(err)
+		}
+		defer f.Close()
```

Comment:
These lines could be extracted out to a small function.

```go
func openFile(filename string) *os.File {
	f, err := os.Open(fileName)
	if err != nil {
		erroredExit(err)
	}
}
```

```suggestion
		f, err := openFile(fileName)
		defer f.Close()
```